src/devices/cpu/gigatron/gigatron.cpp

```diff
@@ -95,36 +97,174 @@ void gigatron_cpu_device::init()
 	m_x = 0;
 	m_y = 0;
 	m_pc = 0;
+	m_npc = (m_pc + 1);
+	m_inReg = 0xFF;
 	state_add(GTRON_AC,        "AC",        m_ac);
 	state_add(GTRON_X,         "X",         m_x);
 	state_add(GTRON_Y,         "Y",         m_y);
 
 	set_icountptr(m_icount);
 }
 
-void gigatron_cpu_device::branchOp(int op, int mode, int bus, int d)
+void gigatron_cpu_device::branchOp(uint8_t op, uint8_t mode, uint8_t bus, uint8_t d)
 {
+	const uint8_t ZERO = 0x80;
+	uint8_t c = 0;
+	uint8_t ac2 = m_ac ^ ZERO;
+	uint16_t base = m_pc & 0xff00;
+	switch (mode) {
+		case 0:
+			c = true;
+			base = m_y << 8;
+			break;
+		case 1:
+			c = (ac2 > ZERO);
+			break;
+		case 2:
+			c = (ac2 < ZERO);
+			break;
+		case 3:
+			c = (ac2 != ZERO);
+			break;
+		case 4:
+			c = (ac2 == ZERO);
+			break;
+		case 5:
+			c = (ac2 >= ZERO);
+			break;
+		case 6:
+			c = (ac2 <= ZERO);
+			break;
+		case 7:
+			c = true;
+			break;
+	}
+	
+	if (c) {
+		uint8_t b = offset(bus, d);
+		m_npc = base | b;
+	}
 }
 
-void gigatron_cpu_device::aluOp(int op, int mode, int bus, int d)
+void gigatron_cpu_device::aluOp(uint8_t op, uint8_t mode, uint8_t bus, uint8_t d)
 {
-	int b;
+	uint8_t b = 0;
 	(void)b;
```

Comment:
This line does nothing except allow the variable to be possibly unused; it's definitely used now.

Reply:
So, I get rid of the b definition entirely?

Reply:
No, I was referring to the `(void)b;` line. It was already redundant, I think.

Reply:
I added that one because of a compiler warning "variable set but unused". The line can be removed now.